src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,237 @@
+
+                foreach (BlittableJsonReaderObject doc in results)
+                {
+                    await queryResult.AddResultAsync(doc, token.Token);
+                }
+                queryResult.Flush();
+            }
+        }
+
+        protected override async ValueTask ExecuteAndWriteIndexQueryStreamEntriesAsync(TransactionOperationContext context, IndexQueryServerSide query, string format, string debug, string[] propertiesArray,
+            string fileNamePrefix, bool ignoreLimit, bool _, OperationCancelToken token)
+        {
+            //from debug
+
+            var (results, _) = await ExecuteQueryAsync(context, query, debug, ignoreLimit, token);
+
+            await using (var writer = GetBlittableQueryResultWriter(format, isDebug: true, context, HttpContext.Response, RequestHandler.ResponseBodyStream(), fromSharded: false, propertiesArray,
+                             fileNamePrefix))
+            {
+                var queryResult = new StreamDocumentIndexEntriesQueryResult(HttpContext.Response, writer, token);
+                
+                foreach (BlittableJsonReaderObject doc in results)
+                {
+                    await queryResult.AddResultAsync(doc, token.Token);
+                }
+                queryResult.Flush();
+            }
+        }
+
+        public class DocumentBlittableLastModifiedComparer : Comparer<BlittableJsonReaderObject>
+        {
+            public override int Compare(BlittableJsonReaderObject x, BlittableJsonReaderObject y)
+            {
+                return y.GetMetadata().GetLastModified().CompareTo(x.GetMetadata().GetLastModified());
+            }
+
+            public static DocumentBlittableLastModifiedComparer Instance = new();
+        }
+    }
+
+    public readonly struct ShardedStreamQueryOperation : IShardedOperation<StreamResult, (IEnumerator<BlittableJsonReaderObject>, StreamQueryStatistics)>
+    {
+        private readonly HttpContext _httpContext;
+        private readonly Func<(JsonOperationContext, IDisposable)> _allocateJsonContext;
+        private readonly IComparer<BlittableJsonReaderObject> _comparer;
+        private readonly Dictionary<int, PostQueryStreamCommand> _queryStreamCommands;
+        private readonly int _skip;
+        private readonly int _take;
+        private readonly CancellationToken _token;
+
+        public ShardedStreamQueryOperation(HttpContext httpContext, Func<(JsonOperationContext, IDisposable)> allocateJsonContext, IComparer<BlittableJsonReaderObject> comparer, Dictionary<int, PostQueryStreamCommand> queryStreamCommands, int skip, int take, CancellationToken token)
+        {
+            _httpContext = httpContext;
+            _allocateJsonContext = allocateJsonContext;
+            _comparer = comparer;
+            _queryStreamCommands = queryStreamCommands;
+            _skip = skip;
+            _take = take;
+            _token = token;
+            ExpectedEtag = null; //TODO stav: what is this
```

Comment:
At glance it looks it's for supporting NotModified. Don't think it's applicable to query streaming since we don't cache those results on the client side anyway.

Reply:
leaving as null then